### DISCUSSION TITLE
How to apply different sortspec to folders and notes?

### DISCUSSION THREAD

**Discussion by pinetreemoonlight** [SebastianMC/obsidian-custom-sort#100] [Q&A] [answered]
Apologies for the dumb question. I want my subfolders to be sorted alphabetically, but all notes to be sorted by creation date. is this possible without specifically naming the subfolders in the yaml? I often change my subfolder structure, so it would be nice if I didn't have to specify the note sorting rule manually.

**Comment by SebastianMC** [comment, marked as answer]
Hi @pinetreemoonlight,

if I get your question correctly, you need the below sorting specification:
```yaml
---
sorting-spec: |
  target-folder: /
  /folders ...
    < a-z
  /:files ...
    > created
---
```
You read it as:
- inside the root folder of the vault (this is specified as `target-folder: /`)
- collect all subfolders in the top (`/folders ...`)
- and order them alphabetically (the indented `  < a-z` order specification)
- then go all files (`/:files ...`)
- ordered by creation date, newer first (the indented `  > created`)

Indentation matters - the two order specifications have to be indented at least one space more than the surrounding lines

At the same time: do I get your question correctly? Because the exact result is produced by the Obsidian standard UI-selectable sorting '_Created time (new to old)_'. Obsidian always puts folders first, in alphabetical order, regardless of what sorting order you select in UI. And the selected order is applied only to files (aka notes).

The obvious advantage of the custom sort plugin is that you can define a different sorting order per each folder. And that could be one of reasons not to stay (only) with the built-in UI-selectable sorting.

**Comment by pinetreemoonlight** [comment]
Thank you so much! It was probably a stupid question. 
I was unaware of the `/:files` syntax (I tried to look at the readme, but I must have missed it). 

In any case, that solves it. Thanks!
